backend/canvas_app_explorer/canvas_lti_manager/manager.py

```diff
@@ -22,6 +43,13 @@ def __init__(self, api_url: str, api_key: str, course_id: int):
         self.course_id: int = course_id
         self.canvas_api: Canvas = Canvas(api_url, api_key)
 
+    @staticmethod
```

Comment:
It seems like this method and all of this exception code before this could just be consolidated into to `exception.py`. Then this could be a method on the CavasHTTPError class and you could call something like error.convert_error().

Was there are reason for having it external to that in the manager.py here?

Reply:
Even if we got the `canvasapi` library updated, I'm not sure I'd want to reference `CanvasException` or its attributes in `views.py`. I think that would defeat the benefits of having an adapter like `CanvasLTIManager` in the first place, which again are a) to make testing easier and b) make it easier to maintain because the manager class could get rewritten with another library. Any future classes could also crosswalk their exceptions to `CanvasHTTPException` (which I think is really just dealing with raw Canvas API data) with a conversion method too.

It would still be nice to have `canvasapi` provide the status codes, it'd mean we could remove the map dictionary I set up. I agree with you that it's not likely we'd need to replace this library in the short term, but I'm trying to learn and try out good/interesting code design practices.

Reply:
I did want to hear your opinion on this, talk through my thinking, that's why I asked for your review 😁

Reply:
I think it's fine, not going to nitpick it. 

I think if I'd done it I'd just have put this `convert_error` as a method in `CanvasHTTPError` rather than in `CanvasLtiManager` here and call `error.convert()` (or even have it part of `__str__`) rather than` self.convert_error(error)`.